# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1207242828 14400
#      Thu Apr 03 13:13:48 2008 -0400
# Branch release-3-0-x
# Node ID 2f659a9c89f2a951bb50023a3ff6f9a46dfe838f
# Parent  de826649dfa23399c1846e69f58f4e1eb84927fa
octave_fcn_handle::subsref: don't call next_subsref

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,13 @@
+2008-04-03  John W. Eaton  <jwe@octave.org>
+
+	* ov-fcn-handle.cc (octave_fcn_handle::subsref):
+	Don't call next_subsref here.
+
 2008-04-03  Jaroslav Hajek <highegg@gmail.com>
 
 	* DLD-FUNCTIONS/sort.cc (xabs, ascending_compare, operator <,
 	descending_compare, operator >): Move definitions ahead of
 	#include "oct-sort.cc".
 
 2008-04-02  John W. Eaton  <jwe@octave.org>
 
diff --git a/src/ov-fcn-handle.cc b/src/ov-fcn-handle.cc
--- a/src/ov-fcn-handle.cc
+++ b/src/ov-fcn-handle.cc
@@ -134,22 +134,18 @@ octave_fcn_handle::subsref (const std::s
 	error ("%s cannot be indexed with %c", typ_nm.c_str (), type[0]);
       }
       break;
 
     default:
       panic_impossible ();
     }
 
-  // FIXME -- perhaps there should be an
-  // octave_value_list::next_subsref member function?  See also
-  // octave_builtin::subsref.
-
-  if (idx.size () > 1)
-    retval = retval(0).next_subsref (nargout, type, idx);
+  // There's no need to call next_subsref here --
+  // octave_function::subsref will handle that for us.
 
   return retval;
 }
 
 bool
 octave_fcn_handle::set_fcn (const std::string &octaveroot, 
 			    const std::string& fpath)
 {
